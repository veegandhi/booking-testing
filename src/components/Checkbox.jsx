import React from "react";
import { connect } from "react-redux";

const Checkbox = ({ checked, updateSelection }) => (
  <input type="checkbox" checked={checked} onChange={updateSelection} />
);

const ConnectedCheckbox = connect(
  (state, ownProps) => ({
    checked: state.selection.includes(ownProps.id)
  }),
  (dispatch, ownProps) => ({
    updateSelection: () =>
      dispatch({ type: "UPDATE_SELECTION", id: ownProps.id })
  })
)(Checkbox);

export default ConnectedCheckbox;
