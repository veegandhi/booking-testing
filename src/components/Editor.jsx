import React from "react";
import { connect } from "react-redux";

const Editor = ({ value, updateValue, updateKey }) => {
  const [isEditing, setIsEditing] = React.useState(false);
  return (
    <td>
      {isEditing ? (
        <input
          value={value}
          onChange={e => updateValue(e.target.value)}
          onBlur={() => setIsEditing(false)}
        />
      ) : (
        <div className={updateKey} onClick={() => setIsEditing(true)}>{value}</div>
      )}
    </td>
  );
};
export { Editor as Component };
const ConnectedEditor = connect(
  null,
  (dispatch, ownProps) => ({
    updateValue: (value, key) =>
      dispatch({
        type: "UPDATE_BOOKING_VALUE",
        id: ownProps.id,
        value,
        key: ownProps.updateKey
      })
  })
)(Editor);

export default ConnectedEditor;
