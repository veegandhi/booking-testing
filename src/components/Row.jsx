import React from "react";
import { connect } from "react-redux";
import { getBookingById } from "../utils";
import Checkbox from "./Checkbox";
import Editor from "./Editor";

const Row = ({ id, product, quantity }) =>
  console.log(id) || (
    <tr key={id}>
      <th scope="row">
        <Checkbox id={id} />
      </th>
      <td className="id">{id.slice(0, 5)}</td>
      <td className="name">{product.name}</td>
      <td className="seller">{product.seller.name}</td>
      <Editor updateKey="quantity" id={id} value={quantity} />
      <td className="row-total">{`$${(quantity * product.rate) / 100}`}</td>
    </tr>
  );

export { Row as Component };

const makeGetBookingsById = (state, ownProps) => {
  const memoizedSelectorById = getBookingById(ownProps.bookingId);

  return memoizedSelectorById(state);
};
const ConnectedRow = connect(makeGetBookingsById)(Row);

export default ConnectedRow;
