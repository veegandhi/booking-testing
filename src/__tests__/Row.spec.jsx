import React from "react";
import _ from "lodash";
import { shallow } from "enzyme";
import {
  createMockStore,
  populateMockStore,
  mountWithStore
} from "../testUtils.jsx";
import { Component as Row } from "../components/Row";

describe("Row", () => {
  let store;

  beforeEach(() => {
    store = createMockStore();
    store.state.selection = ["booking-1"];
    store.state.data = {
      bookings: [
        {
          quantity: 125000,
          startDate: "2020-03-02T23:59:59.999Z",
          endDate: "2020-03-16T00:00:00.000Z",
          productId: "product-1",
          name: "Handcrafted channels",
          id: "booking-1"
        }
      ],
      products: [
        {
          rate: 1400,
          sellerId: "seller-1",
          name: "Medium Desktop Run of Section ATF Interstitial",
          id: "product-1"
        }
      ],
      sellers: [
        {
          id: "seller-1",
          name: "Kertzmann, Hodkiewicz and Koch"
        }
      ]
    };
    populateMockStore(store);
  });

  it("should render Row", () => {
    const wrapper = shallow(
      <Row
        id="booking-1"
        product={{
          rate: 1400,
          sellerId: "seller-1",
          name: "Medium Desktop Run of Section ATF Interstitial",
          id: "product-1",
          seller: {
            id: "seller-1",
            name: "Kertzmann, Hodkiewicz and Koch"
          }
        }}
        quantity={2000}
      />
    );
    // console.log(wrapper.debug())
    expect(wrapper.find(".id")).toHaveLength(1);
  });
  
  it("should render Row", () => {
    const wrapper = mountWithStore(
      <Row
        id="booking-1"
        product={{
          rate: 1400,
          sellerId: "seller-1",
          name: "Medium Desktop Run of Section ATF Interstitial",
          id: "product-1",
          seller: {
            id: "seller-1",
            name: "Kertzmann, Hodkiewicz and Koch"
          }
        }}
        quantity={2000}
      />,
      store
    );
    console.log(wrapper.debug());
    expect(wrapper.find('Row').debug()).toMatchSnapshot();
  });
});
