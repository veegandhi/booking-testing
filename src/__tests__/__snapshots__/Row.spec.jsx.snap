// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render Row 1`] = `
"<Row id=\\"booking-1\\" product={{...}} quantity={2000}>
  <tr>
    <th scope=\\"row\\">
      <Connect(Checkbox) id=\\"booking-1\\">
        <Checkbox id=\\"booking-1\\" checked={false} updateSelection={[Function: updateSelection]}>
          <input type=\\"checkbox\\" checked={false} onChange={[Function: updateSelection]} />
        </Checkbox>
      </Connect(Checkbox)>
    </th>
    <td className=\\"id\\">
      booki
    </td>
    <td className=\\"name\\">
      Medium Desktop Run of Section ATF Interstitial
    </td>
    <td className=\\"seller\\">
      Kertzmann, Hodkiewicz and Koch
    </td>
    <Connect(Editor) updateKey=\\"quantity\\" id=\\"booking-1\\" value={2000}>
      <Editor updateKey=\\"quantity\\" id=\\"booking-1\\" value={2000} updateValue={[Function: updateValue]}>
        <td>
          <div className=\\"quantity\\" onClick={[Function: onClick]}>
            2000
          </div>
        </td>
      </Editor>
    </Connect(Editor)>
    <td className=\\"row-total\\">
      $28000
    </td>
  </tr>
</Row>"
`;
